### PR TITLE
Feature/support core 2.6.0

### PR DESCRIPTION
**Pull request by vinta**
### PR Reminders
---

1. **Contract**: make sure the code follows our convention, ex: naming, explicit returns, etc.; if uncertain, discuss with others

2. **Test**: make sure tests can cover most normal and edge cases; if not, open follow-up tickets. Also, look out for failed tests and fix them!

3. **CHANGELOG.md**: update when external interfaces are changed

4. Workflow: 
    - Github: assign the pr to yourself; if pairing, can merge directly; else, assign someone to help review
    - Asana: assign the corresponding ticket to yourself and leave the pr link on it for easier follow-ups
    - Discord: if someone is mentioned in this pr, tag on Discord

### CODE REVIEW

File: contracts/limitOrder/LimitOrderBook.sol [hunk 1]
```diff
@@ -322,6 +322,15 @@ contract LimitOrderBook is
     }
 
     function _getPriceByRoundId(address baseToken, uint80 roundId) internal view returns (uint256) {
+        // TODO: support perp-curie-contract 2.6.0
```
Comment (twxia) [inline contracts/limitOrder/LimitOrderBook.sol:325]: 2.6.0's V3 doesn't expose rounddata related logic to support historical data because we cannot make sure the price is under our contol (protect).

Which mans right now we only can do something like this:
```
function _getPriceByRoundId(address baseToken, uint80 roundId) internal view returns (uint256) {
  PriceFeedDispatcher priceFeed = PriceFeedDispatcher(IBaseToken(baseToken).getPriceFeed());
    return priceFeed.getDispatchedPrice(0);
}
 ```
 However, I think this is not what we want. 
 
I'd suggest here we remain the same or directly use somewhat interface like, `import { AggregatorV3Interface } from "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";`, to fulfill the goal. 

or maybe build another fallbackable oracle to suit for this case.

Comment (vinta) [reply contracts/limitOrder/LimitOrderBook.sol:325]: @twxia I'm afraid not, the `_getPriceByRoundId()` is for stop loss order to **verify trigger price (not limit price) is matched or not**. It only works when baseToken is using Chainlink, won't work when using Uniswap or Band (that's expected).

However, I think using `AggregatorV3Interface` directly should work. Something like this:

```solidity
function _getPriceByRoundId(address baseToken, uint80 roundId) internal view returns (uint256) {
    PriceFeedDispatcher priceFeedDispatcher = PriceFeedDispatcher(BaseToken(baseToken).getPriceFeed());
    // NOTE: it should be fine that we don't check "priceFeedDispatcher.getStatus() == Chainlink"
    // because roundId is only used for verifying trigger price (not limit price) is matched or not after the user created the stop loss order
    // when trigger price is matched, the stop loss order becomes a regular limit order
    ChainlinkPriceFeed chainlinkPriceFeed = priceFeedDispatcher.getChainlinkPriceFeedV3();
    AggregatorV3Interface aggregator = AggregatorV3Interface(chainlinkPriceFeed.getAggregator());

    // copy from https://github.com/perpetual-protocol/perp-oracle-contract/blob/main/contracts/ChainlinkPriceFeedV2.sol#L46
    // NOTE: aggregator will revert if roundId is invalid (but there might not be a revert message sometimes)
    // will return (roundId, 0, 0, 0, roundId) if round is not complete (not existed yet)
    // https://docs.chain.link/docs/historical-price-data/
    (, int256 price, , uint256 updatedAt, ) = aggregator.getRoundData(roundId);
    // CPF_IP: Invalid Price
    require(price > 0, "CPF_IP");
    // CPF_RINC: Round Is Not Complete
    require(updatedAt > 0, "CPF_RINC");

    return _formatDecimals(uint256(price), chainlinkPriceFeed.decimals(), 18);
}
```

Comment (twxia) [reply contracts/limitOrder/LimitOrderBook.sol:325]: @vinta That's my undersatnding. However, if you do so, you loss the fallback feature. That's why I suggest remain the same or implement a customzied one in the future.

Comment (vinta) [reply contracts/limitOrder/LimitOrderBook.sol:325]: @twxia I don't quite understand what you mean by remain the same. If we don't change the current code, it will break stop loss orders.

The fallback should be something like:

```solidity
function getChainlinkPriceFeed(address baseToken) {
    try {
        PriceFeedDispatcher priceFeedDispatcher = PriceFeedDispatcher(BaseToken(baseToken).getPriceFeed());
        return priceFeedDispatcher.getChainlinkPriceFeedV3();
    } catch {
        return ChainlinkPriceFeed(IBaseToken(baseToken).getPriceFeed());
    }
}

function _getPriceByRoundId(address baseToken, uint80 roundId) internal view returns (uint256) {
    ChainlinkPriceFeed chainlinkPriceFeed = _getChainlinkPriceFeed(baseToken);
    AggregatorV3Interface aggregator = AggregatorV3Interface(chainlinkPriceFeed.getAggregator());

    // copy from https://github.com/perpetual-protocol/perp-oracle-contract/blob/main/contracts/ChainlinkPriceFeedV2.sol#L46
    // NOTE: aggregator will revert if roundId is invalid (but there might not be a revert message sometimes)
    // will return (roundId, 0, 0, 0, roundId) if round is not complete (not existed yet)
    // https://docs.chain.link/docs/historical-price-data/
    (, int256 price, , uint256 updatedAt, ) = aggregator.getRoundData(roundId);
    // CPF_IP: Invalid Price
    require(price > 0, "CPF_IP");
    // CPF_RINC: Round Is Not Complete
    require(updatedAt > 0, "CPF_RINC");

    return _formatDecimals(uint256(price), chainlinkPriceFeed.decimals(), 18);
}
```

**Comment by tienshaoku** [comment]
Note: though I've mocked out the timestamp in `PerpPortal`, I was still seeing flaky test once and had to resolve it by rerunning the test; have to see if it happens again!
